compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -557,6 +584,101 @@ struct QuantizeWeights final : public luci::CircleNodeMutableVisitor<bool>
   }
 };
 
+/**
+ * @brief Quantize const input tensors using min/max of const values
+ */
+void quantize_const_inputs(luci::CircleNode *node, loco::DataType output_type)
+{
+  auto opcode = node->opcode();
+  auto arity = node->arity();
+
+  loco::Node *input_node{nullptr};
+  luci::CircleConst *const_node{nullptr};
+
+  switch (opcode)
+  {
+    case luci::CircleOpcode::CONV_2D:
+    case luci::CircleOpcode::DEPTHWISE_CONV_2D:
+    case luci::CircleOpcode::FULLY_CONNECTED:
+    case luci::CircleOpcode::TRANSPOSE_CONV:
+      // Handled in QuantizeWeights and QuantizeBias
+      break;
+
+    case luci::CircleOpcode::CONCATENATION:
+      // Handled in propagate_concat_quantparam
+      break;
+
+    case luci::CircleOpcode::ARG_MAX:
+    case luci::CircleOpcode::ARG_MIN:
+    case luci::CircleOpcode::MEAN:
+    case luci::CircleOpcode::PAD:
+    case luci::CircleOpcode::REDUCE_ANY:
+    case luci::CircleOpcode::REDUCE_PROD:
+    case luci::CircleOpcode::REDUCE_MAX:
+    case luci::CircleOpcode::REDUCE_MIN:
+    case luci::CircleOpcode::SUM:
+      // The second input of these Ops should not be quantized
+      // Ex: axis, paddings
+      input_node = node->arg(0);
+      const_node = dynamic_cast<luci::CircleConst *>(input_node);
+      if (const_node != nullptr)
+      {
+        if (output_type == loco::DataType::U8)
+        {
+          quant_const(const_node);
+        }
+        else
+        {
+          throw std::runtime_error("NYI data type");
+        }
+      }
+      break;
+
+    case luci::CircleOpcode::ADD:
+    case luci::CircleOpcode::ADD_N:
+    case luci::CircleOpcode::DIV:
+    case luci::CircleOpcode::EQUAL:
+    case luci::CircleOpcode::GREATER:
+    case luci::CircleOpcode::GREATER_EQUAL:
+    case luci::CircleOpcode::LESS:
+    case luci::CircleOpcode::LESS_EQUAL:
+    case luci::CircleOpcode::MAXIMUM:
+    case luci::CircleOpcode::MINIMUM:
+    case luci::CircleOpcode::MUL:
+    case luci::CircleOpcode::NOT_EQUAL:
+    case luci::CircleOpcode::PRELU:
+    case luci::CircleOpcode::SUB:
+      // Quantize all const inputs using their values
+      for (uint32_t i = 0; i < arity; i++)
+      {
+        input_node = node->arg(i);
+        const_node = dynamic_cast<luci::CircleConst *>(input_node);
+        if (const_node != nullptr)
+        {
+          if (output_type == loco::DataType::U8)
+          {
+            quant_const(const_node);
+          }
+          else
+          {
+            throw std::runtime_error("NYI data type");
+          }
+        }
+      }
+      break;
+
+    default:
+      for (uint32_t i = 0; i < arity; i++)
```

Comment:
May be it is worth to list operators with "default" behavior explicitly and throw an exception in the "default:" case.

Reply:
The 'default' behavior to quantize a const tensor would be quantizing with the const values (implemented in `quant_const` function). But it should be applied according to opcode.

I'd like to make this function work with the allow list because I don't fully understand the behavior of all operators. Currently, the listed operators are only the ones I'm familiar with. It will be possible to add operators in this switch statement as needed.

Reply:
What I'm suggesting is to throw exception "by default", rather that quantize all operator inputs. It would allow to catch operators that are not currently white-listed.

Reply:
> What I'm suggesting is to throw exception "by default", rather that quantize all operator inputs. It would allow to catch operators that are not currently white-listed.

I think the current code is doing the same thing. Exception is being thrown in the `default:` case when unsupported Op (Ops not in the whitelist) has a const input. In the default case, inputs are not quantized.

Reply:
I see now, thank you.